ansible_base/lib/dynamic_config/dynamic_urls.py

```diff
@@ -10,12 +10,19 @@
 for url_type in url_types:
     globals()[url_type] = []
 
-for app in getattr(settings, 'INSTALLED_APPS', []):
+installed_apps = getattr(settings, 'INSTALLED_APPS', [])
+oauth2_provider_installed = "ansible_base.oauth2_provider" in installed_apps
+for app in installed_apps:
     if app.startswith('ansible_base.'):
         if not importlib.util.find_spec(f'{app}.urls'):
             logger.debug(f'Module {app} does not specify urls.py')
             continue
         url_module = __import__(f'{app}.urls', fromlist=url_types)
         logger.debug(f'Including URLS from {app}.urls')
         for url_type in ['api_version_urls', 'root_urls', 'api_urls']:
-            globals()[url_type].extend(getattr(url_module, url_type, []))
+            urls = getattr(url_module, url_type, [])
+            if oauth2_provider_installed:
+                from ansible_base.oauth2_provider.utils import attach_oauth2_permission_class_to_urlpatterns
+
+                attach_oauth2_permission_class_to_urlpatterns(urls)
+            globals()[url_type].extend(urls)
```

Comment:
I think I would remove this from here. If people are using DAB apps we expect them to be using dynamic_urls but they technically don't have to. If we are "fixing" views at runtime in this and someone is not using dynamic_urls they will not get this benefit. Instead we should explicitly add the permission to the DAB views and raise CI errors if a new endpoint is missing the permission class.